lemur/static/app/angular/notifications/notification/notification.js

```diff
@@ -42,25 +42,15 @@ angular.module('lemur')
       PluginService.getByType('notification').then(function (plugins) {
         $scope.plugins = plugins;
         _.each($scope.plugins, function (plugin) {
-          if (plugin.slug === $scope.notification.pluginName) {
-            plugin.pluginOptions = $scope.notification.notificationOptions;
+          if (plugin.slug === $scope.notification.plugin.slug) {
+            plugin.pluginOptions = $scope.notification.plugin.pluginOptions;
             $scope.notification.plugin = plugin;
           }
         });
       });
       NotificationService.getCertificates(notification);
     });
 
-    PluginService.getByType('notification').then(function (plugins) {
-      $scope.plugins = plugins;
-      _.each($scope.plugins, function (plugin) {
-        if (plugin.slug === $scope.notification.pluginName) {
-          plugin.pluginOptions = $scope.notification.notificationOptions;
-          $scope.notification.plugin = plugin;
-        }
-      });
-    });
-
```

Comment:
PR looks good!
I was going through older PR to find if this code is needed or may be left error (I believe so). While doing that I came across code in [source.js](https://github.com/Netflix/lemur/blob/master/lemur/static/app/angular/sources/source/source.js#L44) which may need similar fix. Do you think we can fix it the same way?

Reply:
That looks probable. Let me dig into it and see.

Reply:
In testing that, I've discovered another issue: it appears to be impossible to ever change the plugin associated with either a notification or a source. The options can be updated, but not the plugin itself (and attempting to change it puts the notification/source into a bad state). I'll try to figure out how to fix that and update this PR.